### PR TITLE
Conditional CI

### PR DESCRIPTION
**Pull request by MadeByMike**
This adds conditional logic to running tests on Circle CUI.

If looks at the files that have changed will skip tests if not required.

This adds the ground-work for more conditional logic in future where we only test packages that are change.

Current logic:

 - If the PR only contains markdown or media files, skip CI
 - If the PR is one of a few selected packages ignore all tests other than 'simple tests' 

Future PR:
  - Re-write Cypress GraphQL tests to Jest
  -  Report specific time tests take to Circle
  -  Additional skip logic

### CODE REVIEW

**Comment by MadeByMike** [comment]
I don't think this needs a changeset?

File: .circleci/config.yml [hunk 1]
```diff
@@ -59,36 +61,42 @@ x-integration-steps: &integration-steps
     - store_artifacts:
         path: /tmp/screenshots
 
+x-simple-tests: &simple-tests
```
Comment (mitchellhamilton) [inline .circleci/config.yml]: We could move this back to the job now since the thing we tried with this didn't work.

Comment (MadeByMike) [reply .circleci/config.yml]: I actually think it reads nicer as an inclusion like this. Makes everything at the bottom neat and easy to read. What do you think?

Comment (mitchellhamilton) [reply .circleci/config.yml]: Sure - sounds good

**Pull request merged**